tests/ExceptionHandlerTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+use Meisterwerk\Core\Exception\UnexpectedErrorException;
+use Meisterwerk\Core\ExceptionHandler;
+use PHPUnit\Framework\TestCase;
+
+class ExceptionHandlerTest extends TestCase
+{
+    public function testMailingCallbackWithoutThrowOn(): void
+    {
+        $exception = new \Exception('MailingCallBackTestException');
+        $handler = fn () => (print 'mailing callback success');
+
+        $this->expectOutputString('mailing callback success');
+
+        ExceptionHandler::handleWithCare($exception, 'mailingCallbackTest', $handler, false);
+    }
+
+    public function testMailingCallbackWithThrowOn(): void
+    {
+        $exception = new \Exception('MailingCallBackTestException');
+        $handler = fn () => (print 'mailing callback success');
+        $scope = 'mailingCallbackTest';
+
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Unexpected error (' . $scope . ')');
+        $this->expectExceptionCode(0);
+
+        ExceptionHandler::handleWithCare($exception, $scope, $handler);
+    }
+
+    public function testUnexpectedErrorException(): void
+    {
+        $exception = new UnexpectedErrorException('UnexpectedErrorTestException');
```

Comment:
Warum verwendest du hier nicht die `\Exception('MailingCallBackTestException');` wie in den anderen Tests?

Reply:
Finde ich verwirrend...